### DISCUSSION TITLE
Deriving the P-site offsets

### DISCUSSION THREAD

**Discussion by slukowski** [zhpn1024/ribotish#20] [General]
Hi
I'm interested to learn more about the method of calculating the p-site offsets, and wondered if you would mind discussing a little more about this. I read the section in the paper that talks about how the ratio is calculated using the RPF counts at the annotated TISs and the sum of the RPF counts near the annotated TISs, and have a few questions. 

1. How/where are the annotated TISs defined?
2. Are all transcripts used for the metagene profile to derive the p-site, or just a subset?
3. What happens if two p-site values are equally or very close in the number of RPF counts or their ratio? How is the p-site determined in that case? I imagine this could greatly impact which frame the read is assigned to.

Thank you for a great piece of software!
Sam

**Comment by zhpn1024** [comment]
Hi Sam,

Thank you for your interest in Ribo-TISH. I'm glad to discuss the details.

1. The annotated TISs are from gene annotations,such as GTF CDS records, or BED thick regions.
2. All protein coding genes are used, and only one transcript is chosen for each gene.
3. Frame is determined before P-site offset values. P-site offset is restricted by the frame. For example, if the first frame (red) is selected, then the possible offset values are restricted to 9, 12 and 15.

Best,
Peng
